ksql-rest-app/src/main/java/io/confluent/ksql/rest/server/resources/HeartbeatResource.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright 2019 Confluent Inc.
+ *
+ * Licensed under the Confluent Community License (the "License"; you may not use
+ * this file except in compliance with the License. You may obtain a copy of the
+ * License at
+ *
+ * http://www.confluent.io/confluent-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.ksql.rest.server.resources;
+
+import io.confluent.ksql.rest.entity.HeartbeatRequest;
+import io.confluent.ksql.rest.entity.HeartbeatResponse;
+import io.confluent.ksql.rest.entity.HostInfoEntity;
+import io.confluent.ksql.rest.entity.Versions;
+import io.confluent.ksql.rest.server.HeartbeatHandler;
+import io.confluent.ksql.rest.server.HeartbeatHandler.HeartbeatInfo;
+import java.util.TreeMap;
+import javax.ws.rs.POST;
+import javax.ws.rs.Path;
+import javax.ws.rs.Produces;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+import org.apache.kafka.streams.state.HostInfo;
+
+@Path("/heartbeat")
+@Produces({Versions.KSQL_V1_JSON, MediaType.APPLICATION_JSON})
+public class HeartbeatResource {
+
+  private final HeartbeatHandler heartbeatHandler;
+
+  public HeartbeatResource(HeartbeatHandler heartbeatHandler) {
+    this.heartbeatHandler = heartbeatHandler;
+  }
+
+  @POST
+  public Response registerHeartbeat(final HeartbeatRequest request) {
+    handleHeartbeat(request);
+    return Response.ok(new HeartbeatResponse(true)).build();
+  }
+
+  private void handleHeartbeat(final HeartbeatRequest request) {
+    HostInfoEntity hostInfoEntity = request.getHostInfo();
+    HostInfo hostInfo = new HostInfo(hostInfoEntity.getHost(), hostInfoEntity.getPort());
+    long timestamp = request.getTimestamp();
+
+    heartbeatHandler.getReceivedHeartbeats().putIfAbsent(hostInfo, new TreeMap<>());
```

Comment:
You're exposing the internals of the map to this resource.  Instead of returning a mutable map through getReceivedHeartbeats(), why not have a method like registerHeartbeat(hostInfo, timestamp) on heartbeatHandler?